gems/pending/postgres_ha_admin/database_yml.rb

```diff
@@ -0,0 +1,58 @@
+require 'util/miq-password'
+require 'fileutils'
+
+module PostgresHaAdmin
+  class DatabaseYml
+    attr_reader :db_yml_file, :environment
+
+    def initialize(db_yml_file, environment)
+      @db_yml_file = db_yml_file
+      @environment = environment
+    end
+
+    def pg_params_from_database_yml
+      rails_params_to_pg(YAML.load_file(db_yml_file)[environment])
+    end
+
+    def update_database_yml(params)
+      db_yml = YAML.load_file(db_yml_file)
+      db_yml[environment].merge!(pg_parameters_to_rails(params))
+      remove_empty(db_yml[environment])
+
+      new_name = "#{db_yml_file}_#{Time.current.strftime("%d-%B-%Y_%H.%M.%S")}"
+      FileUtils.copy(db_yml_file, new_name)
+      begin
+        File.write(db_yml_file, db_yml.to_yaml)
+      rescue StandardError
+        FileUtils.copy(new_name, db_yml_file)
```

Comment:
I would just make this a move command so that we avoid creating a ton of timestamped yml files if this fails over and over again for the same reason.

I'm okay with keeping the previous yml file as a timestamped version if we succeed though so no need to remove `new_name` if we failover successfully. That way we will be able to track to string of databases we have been connected to if we need to do debugging.